cunumeric/module.py

```diff
@@ -1261,8 +1267,9 @@ def _concatenate(
     # Check to see if we can build a new tuple of cuNumeric arrays
     leading_dim = 0
     cunumeric_inputs = inputs
-
-    leading_dim = common_info.shape[axis] * len(cunumeric_inputs)
+    for arr in cunumeric_inputs:
+        leading_dim += arr.shape[axis]
```

Comment:
can this be rewritten to `leading_dim = sum(arr.shape[axis] for arr in inputs)`?

Reply:
I also think your code w/ generator should work but on prom, `leading_dim` becomes generator for some reason, which makes the ndarray creation fail. 
`leading_dim = sum(list(arr.shape[axis] for arr in inputs))` works w/o the error. On my local machine, your code works.

Reply:
What Python version are you using on prom?

Reply:
It is 3.8.12

Reply:
btw, I think the symbol `sum` being redefined in `module.py` can be problematic later on, so maybe you should make a copy of the builtin sum at the beginning before it gets redefined, just like how some of other builtins are handled here: https://github.com/nv-legate/cunumeric/blob/branch-22.03/cunumeric/module.py#L31-L35

Reply:
can you try the workaround for the builtin sum I mentioned and see if that changes anything?

Reply:
It worked. the sum implementation in `module.py` was used for the statement.

Reply:
yeah that's what I thought. a similar thing tripped me over the other day and I learned it the hard way.

Reply:
Then, I'll add the _builtin_sum code in `module.py` Thanks for saving me out of this issue :)